src/google/protobuf/compiler/parser.cc

```diff
@@ -1373,7 +1373,7 @@ bool Parser::ParseOption(Message* options,
           value_location.AddPath(
               UninterpretedOption::kNegativeIntValueFieldNumber);
           uninterpreted_option->set_negative_int_value(
-              -static_cast<int64>(value));
+              static_cast<int64>(-value));
```

Comment:
http://stackoverflow.com/questions/8026694/c-unary-minus-operator-behavior-with-unsigned-operands says the standard defines it as doing the right thing, and all the Clang and GCC versions I've tried are fine with it. Would you prefer this or `kuint64max - value + 1` (I'm fine with either)?